var/spack/repos/builtin/packages/py-melissa-core/package.py

```diff
@@ -0,0 +1,56 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class PyMelissaCore(PythonPackage):
+    """Melissa is a file-avoiding, adaptive, fault-tolerant and elastic
+    framework, to run large-scale sensitivity analysis or deep-surrogate
+    training on supercomputers.
+    This package builds the launcher and server modules.
+    """
+
+    homepage = "https://gitlab.inria.fr/melissa/melissa"
+    git = "https://gitlab.inria.fr/melissa/melissa.git"
+    maintainers("robcaulk", "mschouler", "raffino")
+
+    version("develop", branch="develop")
+
+    # define variants (DL, DL-Torch, DL-Tensorflow)
+    variant(
+        "MELISSA_DL",
+        default=False,
+        description="Install all Deep Learning requirements (Pytorch and TensorFlow)",
+    )
+    variant(
+        "MELISSA_TORCH",
+        default=False,
+        description="Install Deep Learning requirements with Pytorch only",
+    )
+    variant(
+        "MELISSA_TF",
+        default=False,
+        description="Install Deep Learning requirements with TensorFlow only",
+    )
+
+    depends_on("python@3.8.0:", type=("build", "run"))
+    depends_on("py-setuptools@46.4:", type=("build"))
+    # requirements.txt (SA)
+    depends_on("py-pyzmq@22.3.0:", type="run")
+    depends_on("py-mpi4py@3.1.3:", type="run")
```

Comment:
```suggestion
    depends_on("py-mpi4py@3.1.3", type="run")
```
According to requirements.txt, only 3.1.3 is supported

Reply:
Good point, we will make it a non-strict version dependency in the requirement file.